components/basecomponents/StoryText.tsx

```diff
@@ -18,57 +19,71 @@ const scrollToRef = (ref: MutableRefObject<HTMLParagraphElement | null>, div: Mu
   }
 };
 
-const StoryText = ({ textLanguage, audioLanguage, id, audio, onClick }: StoryTextProps): JSX.Element => {
+const getPosition = (phrase: StoryPhrase, audioLanguage: Language): { end: number; start: number } => {
+  type ObjectKey = keyof typeof phrase;
+  const audioLanguageKey = audioLanguage === 'uk' ? 'uk' : 'cs';
+  const start = `start_${audioLanguageKey}` as ObjectKey;
+  const end = `end_${audioLanguageKey}` as ObjectKey;
+
+  return { start: Number(phrase[start]), end: Number(phrase[end]) };
+};
+
+const StoryText = ({ textLanguage, audioLanguage, audio, onClick }: StoryTextProps): JSX.Element => {
   const phraseRef = useRef<HTMLParagraphElement | null>(null);
   const containerRef = useRef<HTMLDivElement | null>(null);
+  const mainLanguage = getCountryVariant();
+  const {
+    query: { story },
+  } = useRouter();
+
+  const [storyData, setStoryData] = useState<StoryPhrase[]>([]);
+
+  useEffect(() => {
```

Comment:
This works, but maybe it's better practice to retrieve this story data in `getStaticProps` in `[story].tsx`  and then pass it down?

Reply:
updated. phrases loaded with getStaticProps and passed down to component